packages/amplify_api/lib/method_channel_api.dart

```diff
@@ -55,6 +58,61 @@ class AmplifyAPIMethodChannel extends AmplifyAPI {
     return result;
   }
 
+  @override
+  GraphQLSubscriptionOperation<T> subscribe<T>(
+      {@required GraphQLRequest<T> request,
+      @required Function(GraphQLResponse<T>) onData,
+      Function() onEstablished,
+      Function(dynamic) onError,
+      Function() onDone}) {
+    const _eventChannel =
+        EventChannel('com.amazonaws.amplify/api_observe_events');
+    _allSubscriptionsStream =
+        _allSubscriptionsStream ?? _eventChannel.receiveBroadcastStream(0);
+
+    //TODO: Either re-name the cancelToken field to id, or remove entirely
+    final subscriptionId = request.cancelToken;
+
+    _setupSubscription(
+      id: subscriptionId,
+      request: request,
+      onEstablished: onEstablished,
+    );
+
```

Comment:
This *might* be a terrible idea - but *if* we want to return an actual StreamSubscription as a member of the operation class, we create a StreamController separate from the _subscription one, and then relay events from _subscription once all the appropriate parsing has been done.

I guess one issue here would be customers manipulating the returned StreamSubscription ('myStream', for example) and expecting cancelling it to have the same affect as cancelling the operation.

Reply:
Ashish and I talked about this, we need to rearchitect this implementation to later support returning a StreamSubscription more natively. The way we are using streams are
1. Setup using methodChannel
2. Send data through eventChannel
3. Close/cancel using methodChannel

is not ideal where all of these steps can be done using just eventChannels. While calling `receiveBroadcaseStream([arguments])`, you pass your request params(such as GraphQLRequest) and flutter will establish a dedicated stream for you for this graphql subscription.